editor/editor_inspector.cpp

```diff
@@ -3114,7 +3114,24 @@ void EditorInspector::update_tree() {
 			// Build the doc hint, to use as tooltip.
 
 			// Get the class name.
-			StringName classname = doc_name == "" ? object->get_class_name() : doc_name;
+			StringName classname = doc_name;
+			if (classname == "") {
+				if (Object::cast_to<Resource>(object) && object->get_script()) {
+					Resource *res = cast_to<Resource>(object);
```

Comment:
If `cast_to<>()` is the same as `Object::cast_to<>()`, you can pull `Resource *res = cast_to<Resource>(object);` outside the `if`, right?

Reply:
Btw it's a static method and should always be used as `Object::cast_to`. Relying on the fact that `this` happens to be an object to do a method call isn't a good practice here.